### PR TITLE
Synchronise server tags

### LINKED ISSUE #1651: Adding support for custom climbables.
### Feature description

The amount of blocks that are climbable can be extended via a datapack in climbables.json. However this results in false simulation checks with Grim.

I have set chains to be climbable via a datapack, climbing the chain results in falses from Grim. The default vanilla climbables work fine.

I would like to see support for using more blocks as climbables. If this is not possible due to not being able to read what blocks are climbable, perhaps let the user define climbable blocks in the config?

### PR DESCRIPTION
**Pull request by SamB440**
This adds support for synchronisation of server tags, currently only for the `climbables` tag.

Fixes #1651

### CODE REVIEW

File: src/main/java/ac/grim/grimac/events/packets/PacketServerTags.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+public class PacketServerTags extends PacketListenerAbstract {
+
+    @Override
+    public void onPacketSend(PacketSendEvent event) {
+        if (event.getPacketType() == PacketType.Play.Server.TAGS || event.getPacketType() == PacketType.Configuration.Server.UPDATE_TAGS) {
+            GrimPlayer player = GrimAPI.INSTANCE.getPlayerDataManager().getPlayer(event.getUser());
+            if (player == null) return;
+
+            WrapperPlayServerTags tags = new WrapperPlayServerTags(event);
+            player.tagManager.handleTagSync(tags);
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/events/packets/PacketServerTags.java]: Shouldn't this be latency compensated?

Comment (SamB440) [reply src/main/java/ac/grim/grimac/events/packets/PacketServerTags.java]: I don't think it matters, in most cases the tags are going to be sent during config stage, in which case you're not even in the game yet, and for play stage well I think that's a very rare occurrence and a minor false wouldn't matter as servers shouldn't be spamming that packet

Comment (ManInMyVan) [reply src/main/java/ac/grim/grimac/events/packets/PacketServerTags.java]: We should still lag compensate it, some servers might, for example, make some climbables permission based.

File: src/main/java/ac/grim/grimac/utils/data/tags/SyncedTags.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * This class stores tags that the client is aware of.
+ */
+public final class SyncedTags {
+
+    private static final ResourceLocation BLOCK = ResourceLocation.minecraft("block");
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/utils/data/tags/SyncedTags.java]: this is "blocks" in pre 1.21

Comment (SamB440) [reply src/main/java/ac/grim/grimac/utils/data/tags/SyncedTags.java]: Did the climbable tag name change at all as well?

Comment (ManInMyVan) [reply src/main/java/ac/grim/grimac/utils/data/tags/SyncedTags.java]: I don't think so

**Pull request merged**